code/modules/fish/fish_tank.dm

```diff
@@ -0,0 +1,706 @@
+	set name = "Toggle Tank Lid"
+	set category = "Object"
+	set src in view(1)
+	toggle_lid(usr)
+
+/obj/machinery/fishtank/proc/toggle_lid(var/mob/living/user)
+	lid_switch = !lid_switch
+	update_icon()
+
+/obj/machinery/fishtank/verb/toggle_light_verb()
+	set name = "Toggle Tank Light"
+	set category = "Object"
+	set src in view(1)
+	toggle_light(usr)
+
+/obj/machinery/fishtank/proc/toggle_light(var/mob/living/user)
+	light_switch = !light_switch
+	if(light_switch)
+		set_light(2,2,"#a0a080")
+	else
+		set_light(0)
+
+//////////////////////////////
+//		NEW() PROCS			//
+//////////////////////////////
+
+/obj/machinery/fishtank/New()
+	..()
+	fish_list = new/list()
+	egg_list = new/list()
+	if(!has_lid)				//Tank doesn't have a lid/light, remove the verbs for then
+		verbs -= /obj/machinery/fishtank/verb/toggle_lid_verb
+		verbs -= /obj/machinery/fishtank/verb/toggle_light_verb
+
+/obj/machinery/fishtank/tank/New()
+	..()
+	if(prob(5))					//5% chance to get the castle decoration
+		icon_state = "tank2"
+
+//////////////////////////////
+//		ICON PROCS			//
+//////////////////////////////
+
+/obj/machinery/fishtank/update_icon()
+	overlays.Cut()
+
+	//Update Alert Lights
+	if(has_lid)											//Skip the alert lights for aquariums that don't have lids (fishbowls)
+		if(egg_count > 0)								//There is at least 1 egg to harvest
+			overlays += "over_egg"
+		if(lid_switch == 1)								//Lid is closed, lid status light is red
+			overlays += "over_lid_1"
+		else											//Lid is open, lid status light is green
+			overlays += "over_lid_0"
+		if(food_level > 5)								//Food_level is high and isn't a concern yet
+			overlays += "over_food_0"
+		else if(food_level > 2)							//Food_level is starting to get low, but still above the breeding threshold
+			overlays += "over_food_1"
+		else											//Food_level is below breeding threshold, or fully consumed, feed the fish!
+			overlays += "over_food_2"
+		overlays += "over_leak_[leaking]"				//Green if we aren't leaking, light blue and slow blink if minor link, dark blue and rapid flashing for major leak
+
+	//Update water overlay
+	if(water_level == 0) return							//Skip the rest of this if there is no water in the aquarium
+	var/water_type = "_clean"							//Default to clean water
+	if(filth_level > 5)	water_type = "_dirty"			//Show dirty water above filth_level 5 (breeding threshold)
+	if(water_level > (water_capacity * 0.85))			//Show full if the water_level is over 85% of water_capacity
+		overlays += "over_[tank_type]_full[water_type]"
+	else if(water_level > (water_capacity * 0.35))		//Show half-full if the water_level is over 35% of water_capacity
+		overlays += "over_[tank_type]_half[water_type]"
+
+	return
+
+//////////////////////////////
+//		PROCESS PROC		//
+//////////////////////////////
+
+//Stops atmos from passing wall tanks, since they are effectively full-windows.
+///obj/machinery/fishtank/wall/CanAtmosPass(var/turf/T) paradise code yo
+	return 0
```

Comment:
holy fuck no why

Reply:
This is clearly meant to be commented out.